Improve BackupQuiz Account Key Fetching

aaronmgdr opened the pull request:
### Description

🔐 fetch account key mneumonic directly from store so it doesnt get passed thru nav params

### Tested

changes to tests are because it is now async. 

### Backwards compatibility

yes

Commit:
start analytics

Commit:
use existing subtracking tracking

Commit:
Merge branch 'master' into aaronmgdr/account-key-analytics

# Conflicts:
#	packages/mobile/src/backup/BackupPhrase.tsx
#	packages/mobile/src/backup/BackupQuiz.tsx

Commit:
add test for word press and fix
fix top spacing
improve tracking

Commit:
not used since using sub events

Commit:
cleanup unused constants
FIX a wrong constant

dedeuplicate code

aaronmgdr committed:
Merge branch 'master' into aaronmgdr/account-key-analytics

Commit:
fix test

Commit:
🔐 fetch account key mneumonic directly from store so it doesnt get passed thru nav params

jmrossy commented on packages/mobile/src/backup/BackupQuiz.tsx:
```diff
@@ -82,21 +84,32 @@ export class BackupQuiz extends React.Component<Props, State> {
-    const mnemonic = this.getMnemonicFromNavProps()
+  componentDidMount = async () => {
+    const mnemonic = await this.retrieveMnemonic()
     const shuffledMnemonic = getShuffledWordSet(mnemonic)
```
Should you check for null/empty here since that's what the method returns on error

aaronmgdr replied:
hmm a good question. I wasn't sure exactly how to handle the empty case. right now retrieveMnemonic returns an empty string if there is an error.

I guess it probably is better to short circuit if there isnt a mneumonic aye?

aaronmgdr replied:
ok looking that the code more it seems very very very unlikely mnemonic will ever be blank. we already Log and throw errors in several places when it is. I don't think there is much to be gained from doing an empty string check here.

jmrossy commented on packages/mobile/test/values.ts:
```diff
@@ -13,6 +13,7 @@ import {
   RecipientWithContact,
   RecipientWithMobileNumber,
 } from 'src/recipients/recipient'
+import { Screens } from 'src/navigator/Screens'
```
Why dis?

aaronmgdr replied:
mistake :)

Commit:
unused

Commit:
Merge branch 'master' into aaronmgdr/move-account-key

Commit:
Merge branch 'master' into aaronmgdr/move-account-key

# Conflicts:
#	packages/mobile/src/analytics/constants.ts
#	packages/mobile/src/backup/BackupQuiz.tsx

i1skn commented on packages/mobile/src/backup/BackupQuiz.tsx:
```diff
@@ -82,21 +84,32 @@ export class BackupQuiz extends React.Component<Props, State> {
+        title: message,
+        component: 'BackupQuiz',
+      })
+      return ''
```
I do not understand why we return `''` in case of an error?

aaronmgdr replied:
basically that was just to get typescript to not complain about type being undefined. 

would love to talk about handling the error case cause at the moment feels like there are a lot of `if`s and `trys` and its getting out of hand

Commit:
only fetch account key on mount

Commit:
do not try

add error handle to useAccountKey

jmrossy reviewed (approved):
lgtm :)

Commit:
dry out the code
dont catch in the components as the only thing we are catching is the throw we just did

Commit:
Merge branch 'master' into aaronmgdr/move-account-key

# Conflicts:
#	packages/mobile/src/backup/BackupPhrase.tsx

Commit:
YOLO

mergify[bot] committed:
Merge branch 'master' into aaronmgdr/move-account-key

mergify[bot] committed:
Merge branch 'master' into aaronmgdr/move-account-key

mergify[bot] committed:
Merge branch 'master' into aaronmgdr/move-account-key

mergify[bot] committed:
Merge branch 'master' into aaronmgdr/move-account-key

Commit:
bye

mergify[bot] committed:
Merge branch 'master' into aaronmgdr/move-account-key

mergify[bot] committed:
Merge branch 'master' into aaronmgdr/move-account-key

mergify[bot] committed:
Merge branch 'master' into aaronmgdr/move-account-key

mergify[bot] committed:
Merge branch 'master' into aaronmgdr/move-account-key

mergify[bot] committed:
Merge branch 'master' into aaronmgdr/move-account-key

Commit:
remove duplicate test from bad merge

mergify[bot] committed:
Merge branch 'master' into aaronmgdr/move-account-key
